src/main/java/org/stellar/sdk/Address.java

```diff
@@ -0,0 +1,170 @@
+   */
+  public static Address fromAccount(byte[] accountId) {
+    return new Address(StrKey.encodeStellarAccountId(accountId));
+  }
+
+  /**
+   * Creates a new {@link Address} from a Stellar Contract ID.
+   *
+   * @param contractId the byte array of the Stellar Contract ID.
+   * @return a new {@link Address} object from the given Stellar Contract ID.
+   */
+  public static Address fromContract(byte[] contractId) {
+    return new Address(StrKey.encodeContractId(contractId));
+  }
+
+  /**
+   * Creates a new {@link Address} from a {@link SCAddress} XDR object.
+   *
+   * @param scAddress the {@link SCAddress} object to convert
+   * @return a new {@link Address} object from the given XDR object
+   */
+  public static Address fromSCAddress(SCAddress scAddress) {
+    switch (scAddress.getDiscriminant()) {
+      case SC_ADDRESS_TYPE_ACCOUNT:
+        return new Address(StrKey.encodeStellarAccountId(scAddress.getAccountId()));
+      case SC_ADDRESS_TYPE_CONTRACT:
+        return new Address(StrKey.encodeContractId(scAddress.getContractId().getHash()));
+      default:
+        throw new IllegalArgumentException("Unsupported address type");
+    }
+  }
+
+  /**
+   * Creates a new {@link Address} from a {@link SCVal} XDR object.
+   *
+   * @param scVal the {@link SCVal} object to convert
+   * @return a new {@link Address} object from the given XDR object
+   */
+  public static Address fromSCVal(SCVal scVal) {
+    if (!SCValType.SCV_ADDRESS.equals(scVal.getDiscriminant())) {
+      throw new IllegalArgumentException("SCVal is not of type SCV_ADDRESS");
+    }
+    return Address.fromSCAddress(scVal.getAddress());
+  }
+
+  /**
+   * Converts this object to its {@link SCAddress} XDR object representation.
+   *
+   * @return a new {@link SCAddress} object from this object
+   */
+  public SCAddress toSCAddress() {
+    SCAddress scAddress = new SCAddress();
+    switch (this.type) {
+      case ACCOUNT:
+        scAddress.setDiscriminant(org.stellar.sdk.xdr.SCAddressType.SC_ADDRESS_TYPE_ACCOUNT);
+        scAddress.setAccountId(KeyPair.fromPublicKey(this.key).getXdrAccountId());
+        break;
+      case CONTRACT:
+        scAddress.setDiscriminant(org.stellar.sdk.xdr.SCAddressType.SC_ADDRESS_TYPE_CONTRACT);
+        scAddress.setContractId(new Hash(this.key));
+        break;
+      default:
+        throw new IllegalArgumentException("Unsupported address type");
+    }
+    return scAddress;
+  }
+
+  /**
+   * Converts this object to its {@link SCVal} XDR object representation.
+   *
+   * @return a new {@link SCVal} object from this object
+   */
+  public SCVal toSCVal() {
+    SCVal scVal = new SCVal();
+    scVal.setDiscriminant(SCValType.SCV_ADDRESS);
+    scVal.setAddress(this.toSCAddress());
+    return scVal;
+  }
+
+  /**
+   * Returns the byte array of the Stellar public key or contract ID.
+   *
+   * @return the byte array of the Stellar public key or contract ID.
+   */
+  public byte[] getKey() {
```

Comment:
maybe `getBytes()` for a more general indication? since an address is just an opaque value, not necessarily key info.

Reply:
Fixed in 0a473e72d4d0f446bc8b2f399b623096b1f599fa